### ISSUE TITLE
F-Droid release

### ISSUE THREAD

**Issue by 0verEngineer** [javulticat/NewPipe#13]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/javulticat/NewPipe/issues) or [closed](https://github.com/javulticat/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/javulticat/NewPipe/blob/sponsorblock/.github/CONTRIBUTING.md).

### Feature description

Would a F-Droid release be possible?

### Why do you want this feature?

Ease of use

### Why is the feature relevant to this fork?

..

**Comment by javulticat** [comment]
### TL;DR

This project already fully supports F-Droid builds/releases; it just needs to be accepted into an F-Droid repo that will build/host F-Droid releases of it. Those among the community who want F-Droid releases to become available, please open an Issue with your favorite F-Droid repo(s) (e.g., [IzzyOnDroid](https://gitlab.com/IzzyOnDroid/repo/-/issues), [F-Droid official](https://gitlab.com/fdroid/fdroiddata/-/issues)) requesting that they add this project to their repo. Remember to add links in this thread to any request opened so that others who are interested can upvote/comment on it to indicate how much demand there is for F-Droid releases among our community - the more demand from its community, the better a project's chances of getting added to a repo (this will also help avoid the accidentally creation of duplicate requests in the same F-Droid repo).

While I am asking that the community makes these requests (largely because these requests are more likely to be successful if they are backed by members of a project's community - ideally many members - rather than just its author/maintainer), I will gladly support them in any way I can, such as by providing additional info or making code changes that repo maintainers may require before allowing this project to be added to their repo.

Alternatively (or in the interim), consider using an app manager that can install/update apps directly from GitHub (e.g., [Obtainium](https://github.com/ImranR98/Obtainium#readme), which also supports F-Droid releases), as I've made changes to this repo which should result in an experience that's relatively similar to F-Droid when using an app manager that supports GitHub as a source.

___

Hi there, good question! So, yes, this project already has the necessary infrastructure in place to support being released on F-Droid. However, the issue is getting it added to a public F-Droid repository that will build and distribute/release it on F-Droid. I fully endorse that endeavor, but I humbly ask that the members of the community, such as yourself, who are interested in having this project available via F-Droid, to please be the ones that create the request(s) (and/or add your support to existing requests once they are created) for this project to be added to any F-Droid repos. All it takes is opening an Issue (very similar to this one), but within the F-Droid repos themselves.

Not only would this help me out by taking something off of my plate, but, more importantly, it would likely also significantly improve the chances of success for any request. These requests tend to carry a lot more weight if they come from a project's community (and, ideally, have many members of that community showing their support through upvotes/comments on the request) rather than from its author(s)/maintainer(s).

I would recommend starting with making a request to include the project in [IzzyOnDroid's repo](https://gitlab.com/IzzyOnDroid/repo/-/issues) (which is where `polymorphicshade`'s fork was hosted), though [the official F-Droid repo](https://gitlab.com/fdroid/fdroiddata/-/issues) (which currently hosts upstream NewPipe) could also be tried (or any other public F-Droid repository the community trusts).

FYI, the official repo has more stringent inclusion criteria than IzzyOnDroid's repo, so that will likely be a longer/more difficult path. That's why many projects start out on IzzyOnDroid while they wait on potentially getting accepted to the official F-Droid repo. I am not sure if `polymorphicshade`'s fork was only available on IzzyOnDroid because the official repo rejected it, or it went to IzzyOnDroid first and then just never tried to get added to the official repo.

If/when someone opens a request with an F-Droid repo, please post a link to it in this thread so the rest of the community that would like for this project to be available via F-Droid will know where they can show their support (and to help ensure that we are following good etiquette and only opening one request per F-Droid repo).

For my part, I will monitor any request made and will be more than happy to provide any additional information and/or work on any changes to the code that F-Droid repo maintainers may require before the project can be added to their repo. Feel free to tag me in any request (my [GitLab account](https://gitlab.com/javulticat) is also @javulticat), but, as long as a link gets posted in this thread, I'll be able to track it.

For full transparency, I'll also note that it's possible for a project to self-host their own private F-Droid repository using their own servers, but I would strongly prefer not pursuing that option (due to the added complexity, time, and ongoing costs it would take to set up and maintain) unless it was the last resort (i.e., for some reason, the project gets rejected by all major public repositories).

I'll also note that there are app managers (like [Obtainium](https://github.com/ImranR98/Obtainium#readme)) that can install/update apps both from F-Droid repos and directly from GitHub/GitLab/etc. repos, and a unique thing I've done with this fork (that you won't find in either upstream NewPipe or `polymorphicshade`'s fork) is the creation of a fully automated release publishing system directly in GitHub (#4), which was intended to enable this GitHub repo to function relatively similar to a private F-Droid repo (when used with an app manager that supports GitHub as a source). Some may find this to be a viable alternative to having the project available via F-Droid if they are mainly looking for a way to be able to install/update the app without needing to manually download and install APKs from our [Releases](https://github.com/javulticat/NewPipe/releases) page.

Thanks for your understanding and interest in this project!

**Comment by 0verEngineer** [comment]
Hey, i would do it, no problem but i think for this to be successful the app needs its own name.. @javulticat

**Comment by javulticat** [comment]
@0verEngineer Thanks, glad to hear it!

As far as a potential rename/rebrand, it's something I initially considered, but decided that maintaining the same name/branding to start with (this repo is still barely a week old) would help with the transition for the community coming from `polymorphicshade`'s fork, by making it very clear that this project is intended to be a direct continuation of that work.

However, I hear you, and so I will plan on creating a poll to solicit feedback on a potential rename and/or rebrand once our community has had a bit more time to grow and coalesce.

### Behind the scenes

Here's a rough draft (very rough, I'm _not_ a designer) of a logo idea I had quickly thrown together when I was initially exploring the possibility of rebranding the project. This would have been for a name change to something like `NewPipeX` (a play on the full "NewPipe **x** SponsorBlock **x** Return YouTube Dislike" name used in the README):

![image](https://github.com/javulticat/NewPipe/assets/31746850/7fad9f28-d3be-4558-8b41-1ce1623e96e0)

(Don't worry, if the community does end up wanting the project to be rebranded, I will get a professional to create any new design assets that are needed 😅)

**Comment by 0verEngineer** [comment]
Ok nice, i think we need rebranding because an app with this name already exists in F-Droid, once the rebranding is done i create a feature request on the official F-Droid repo.

I like the name NewPipeX and the logo idea as well.

**Comment by ajayyy** [comment]
@javulticat FYI, upstream NewPipe devs were very clear that they didn't want the word "NewPipe" in the fork's official name

**Comment by 0verEngineer** [comment]
@ajayyy Have you links to discussions about this?

And imho they can't do anything about a forks name so...

**Comment by ajayyy** [comment]
> And imho they can't do anything about a forks name so...

If you are fine being an a**hole, anything is possible :P

**Comment by 0verEngineer** [comment]
> If you are fine being an a**hole, anything is possible :P

This does not make me or the repo owner to an a**hole, the name is not protected and forks should have a name that the user will recognize, especially if the project does plan on staying up to date with the origin project. It is a far more a**holish behavior to try to restrict this stuff. This is only one more weird thing the NewPipe devs have decided.

I did also ask you about a link where this is stated because i want to read why they would trying to do something like this.

Also every other Fork of NewPipe i know has `NewPipe` in its Name, as it should be.

**Comment by SyCoREAPER** [comment]
PipeTube - SponsorBlock x Return YouTube Dislike

**Comment by javulticat** [comment]
Temporarily locking this Issue to encourage discussion unrelated to F-Droid support to migrate to the appropriate Discussion thread.  Thanks!

**Comment by javulticat** [comment]
After further consideration, I believe we have arrived at a "chicken-or-the-egg"-type problem here.  I wanted to gather input from the community on rebranding, but, so far, I have received little input (and certainly no consensus).  I believe this is to be expected from a community as new/small as ours currently is, especially since this project is still unknown to the _vast majority_ of those who used `polymorphicshade`'s fork.

And it's almost certainly unknown to _nearly all_ of the upstream `NewPipe` community, which I imagine now contains many former `polymorphicshade` users who have (begrudgingly, perhaps) switched back to using upstream `NewPipe` for the time being because they saw `polymorphicshade`'s repo was archived and are, unfortunately, unaware that this project picked up from where that project left off.

Because of the general lack of awareness of this project's existence at this early stage, any choice we attempt to make as a community right now will suffer from having a small sample size due to the current size of our community.  In order to grow our community enough to avoid that problem, this project needs to become more well-known.  (PS. if you use this project and would like to support it, simply letting people know that it, ya know, exists... would be very appreciated 🙃)

But, while word-of-mouth certainly helps and is very appreciated, having the app available on F-Droid would likely _drastically_ increase its reach/audience and accelerate our community's growth.  But, to ensure we can make the app become available on F-Droid, it likely needs to be rebranded first.  To rebrand it effectively as a community, we need to first grow our community.  In order to grow our community, having the app available on F-Droid would likely _drastically_ increase its reach/audience...

As you can see, we are stuck in a loop.

So, I've decided to break the loop.  To do so, **I will work to make the app become available on F-Droid as soon as possible.**

To that end, the app will undergo a (temporary?) rebranding, incorporating some of the limited feedback I've received from the community.  The app will be renamed to **`NewPipe X`** and will undergo very simple, yet distinctive, graphical tweaks (e.g., changing the color of the logo, etc.), in an attempt to balance the desire to recognize (and _be_ recognized by) our `NewPipe` and `polymorphicshade` lineage, while sufficiently distinguishing this project from both.

Progress related to all of this can be tracked in detail by following the [F-Droid availability project](https://github.com/users/javulticat/projects/2) I've created for that purpose.

Once the app is available on F-Droid, my hope is that it will help the app to rapidly become more well-known and allow our community to grow significantly (and, perhaps most importantly, will inform users that they have another choice besides an unmaintained version of `NewPipe SponsorBlock` that's missing bugfixes and upstream `NewPipe` that's missing key functionality they used: `NewPipe X`, the best of both worlds 🙃).  Once our community on GitHub has grown to the point where we no longer have to worry about small sample sizes, the app's branding can certainly be reconsidered if the community feels it's a priority to do so.

**TL;DR: Rebrand -> get on F-Droid -> gain users -> grow community -> enhance ability to make choices more democratically -> re-rebrand (if necessary)**
